pkg/odo/cli/component/ui/ui.go

```diff
@@ -0,0 +1,212 @@
+package ui
+
+import (
+	"fmt"
+	"github.com/golang/glog"
+	"github.com/redhat-developer/odo/pkg/catalog"
+	"github.com/redhat-developer/odo/pkg/occlient"
+	"github.com/redhat-developer/odo/pkg/odo/cli/ui"
+	"gopkg.in/AlecAivazis/survey.v1"
+	"sort"
+	"strings"
+)
+
+// SelectComponentType lets the user to select the builder image (name only) in the prompt
+func SelectComponentType(options []catalog.CatalogImage) string {
+	var componentType string
+	prompt := &survey.Select{
+		Message: "Which component type would wish to create",
+		Options: getComponentTypeNameCandidates(options),
+	}
+	err := survey.AskOne(prompt, &componentType, survey.Required)
+	ui.HandleError(err)
+	return componentType
+}
+
+func getComponentTypeNameCandidates(options []catalog.CatalogImage) []string {
+	result := make([]string, len(options))
+	for i, option := range options {
+		result[i] = option.Name
+	}
+	sort.Strings(result)
+	return result
+}
+
+// SelectImageTagInteractively lets the user to select a specific tag for the previously selected builder image in a prompt
+func SelectImageTagInteractively(options []catalog.CatalogImage, selectedComponentType string) string {
+	var tag string
+	prompt := &survey.Select{
+		Message: fmt.Sprintf("Which version of type %s  would you wish to create", selectedComponentType),
```

Comment:
Wording is a little awkward here. Also, extra space after `%s`

Reply:
"Which version of '%s' component type do you wish to use"?

Reply:
Did you remove the extra space?

Reply:
Nope, forgot to do that. I will do that now and also use the message you propose

Reply:
Fixed